### PR TITLE
fix: Potential TransformGroup memory leak

### PR DESCRIPTION
**Pull request by workgroupengineering**
## What does the pull request do?

Subscribe/Unsubscribe of the notification Children change on OnPropertyChanged

## What is the current behavior?

- Subscribe the notification on Children change only `ctor`.
- Unsubscribe of the notification Children change is never made.
- If assing Children property does not unsubscribe of the notification of the change of children.

## What is the updated/expected behavior with this PR?

I subscribe only `PropertyChanged` instead of `CollectionChanged`, because when Add or Remove item `AvaloniaList<T>` notify change of `Count` property.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/avalonia-docs with user documentation

### CODE REVIEW

**Comment by jp2masa** [comment]
> I subscribe only PropertyChanged instead of CollectionChanged, because when Add or Remove item AvaloniaList<T> notify change of Count property.

I don't think this assumption is safe. It may be with the current implementation, but replacing items does not change item count.

**Comment by workgroupengineering** [comment]
> > I subscribe only PropertyChanged instead of CollectionChanged, because when Add or Remove item AvaloniaList notify change of Count property.
> 
> I don't think this assumption is safe. It may be with the current implementation, but replacing items does not change item count.

Good catch. I incorrectly assumed Avalonia<T> was like `ObservableCollection<T>` which reports [OnIndexerPropertyChanged](https://github.com/dotnet/runtime/blob/85d3773b9316bc4283bd85eb7e0da76b41b19ee8/src/libraries/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs#L139).

What do you suggest I do?

**Review by MrJul — CHANGES_REQUESTED** [review]
Thanks for the PR!

For correctness `Transforms` should also override all other methods modifying the list: `RemoveAt()`, `RemoveAll`, `InsertAt`, `InsertRange`, etc.
That's a bit much, we already have the nice `ForEachItem` which handles all that, used previously.

The changes can be simplified to:
 - Keep the `IDisposable` returned by `ForEachItem` in a field.
 - When when the `Children` collection changes:
    - Dispose the old `IDisposable`.
	- Unsubcribe from the items in the old collection.
	- Call `ForEachItem` again on the new collection.

This should handle all cases.

File: src/Avalonia.Base/Media/TransformGroup.cs [hunk 1]
```diff
@@ -11,25 +12,17 @@ public sealed class TransformGroup : Transform
         public static readonly StyledProperty<Transforms> ChildrenProperty =
             AvaloniaProperty.Register<TransformGroup, Transforms>(nameof(Children));
 
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("AvaloniaProperty", "AVP1012", 
+        private IDisposable? _childrenNofiticationSubscription = default;
+        private readonly EventHandler ChildTransform_Changed_Handler;
```
Comment (MrJul) [inline src/Avalonia.Base/Media/TransformGroup.cs:16]: Naming: please respect conventions, use something like `_childTransformChangedHandler` or `_onChildTransformChanged`

**Review by MrJul — CHANGES_REQUESTED** [review]
Logic looks good, a couple naming changes left then we're good to go :)

**Review by MrJul — APPROVED** [review]
LGTM, thanks!

**Pull request merged**